libs/blocks/global-navigation/global-navigation.js

```diff
@@ -68,69 +71,144 @@ class Gnav {
         { Search },
       ] = await Promise.all([
         loadBlock('./delayed-utilities.js'),
         loadBlock('./blocks/navMenu/menu.js'),
         loadBlock('./blocks/appLauncher/appLauncher.js'),
         loadBlock('./blocks/profile/profile.js'),
         loadBlock('./blocks/search/gnav-search.js'),
+        loadStyles('profile/profile.css'),
         loadStyles('navMenu/menu.css'),
         loadStyles('search/gnav-search.css'),
       ]);
       this.menuControls = new MenuControls();
       this.decorateMenu = decorateMenu;
       this.decorateLargeMenu = decorateLargeMenu;
       this.appLauncher = appLauncher;
-      this.profile = profile;
+      this.Profile = Profile;
       this.search = Search;
-
-      this.imsReady
-        .then(({ blockEl, profileEl }) => {
-          this.decorateProfileMenu(blockEl, profileEl);
-        });
       resolve();
     });
     return this.ready;
   };
 
-  decorateProfileMenu = async (blockEl, profileEl) => {
+  loadIMS = () => {
+    const { locale, imsClientId, env } = getConfig();
+    if (!imsClientId) return null;
+    // TODO-1 scopes should be defineable by the consumers
+    // We didn't have a use-case for that so far
+    // TODO-2 we should emit an event after the onReady callback
+    window.adobeid = {
+      client_id: imsClientId,
+      scope: 'AdobeID,openid,gnav',
+      locale: locale || 'en-US',
+      autoValidateToken: true,
+      environment: env.ims,
+      useLocalStorage: false,
+      onReady: () => this.decorateProfile(),
+    };
+    const imsScript = document.querySelector('script[src$="/imslib.min.js"]') instanceof HTMLElement;
+    if (!imsScript && !window.adobeIMS) {
+      loadScript('https://auth.services.adobe.com/imslib/imslib.min.js');
+    }
+    return null;
+  };
+
+  decorateProfile = async () => {
+    const { blockEl, decoratedEl } = this.blocks.profile;
+    if (!blockEl) return null;
+
     const accessToken = window.adobeIMS.getAccessToken();
-    if (accessToken) {
-      const { env } = getConfig();
-      const ioResp = await fetch(`https://${env.adobeIO}/profile`, { headers: new Headers({ Authorization: `Bearer ${accessToken.token}` }) });
-
-      if (ioResp.status === 200) {
-        this.profile(blockEl, profileEl, this.menuControls.toggleMenu, ioResp);
-        const appLauncherBlock = this.body.querySelector('.app-launcher');
-        if (appLauncherBlock) {
-          this.appLauncher(profileEl, appLauncherBlock, this.menuControls.toggleMenu);
-        }
-      } else {
-        this.decorateSignIn(blockEl, profileEl);
+    const { env } = getConfig();
+    const profileRes = accessToken
+      ? await fetch(`https://${env.adobeIO}/profile`, { headers: new Headers({ Authorization: `Bearer ${accessToken.token}` }) })
+      : {};
+    if (profileRes.status !== 200) return this.decorateSignIn();
+
+    if (blockEl.children.length > 1) decoratedEl.classList.add('has-menu');
+    decoratedEl.closest('nav.gnav').classList.add('signed-in');
```

Comment:
Maybe it's worth checking whether the closest `nav.gnav` is found before altering it.